### PR TITLE
CI: use Github actions for static tests instead of Travis

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR adds a new github action to run the static test (finally found how to do it) and removes the Travis CI configuration file.

For those interested, here are some of [my tries on my fork](https://github.com/aabadie/RIOT/actions?query=workflow%3Astatic-tests), the [last one](https://github.com/aabadie/RIOT/actions?query=workflow%3Astatic-tests) corresponds to what is in this PR (except a flip between 2 steps)/

### Testing procedure

- The new github action should perform the static test with success. Static test that are expected to fail should fail ("PR needs squashing" for instance)
- Travis CI should not appear anymore

### Issues/PRs references

None

### CODE REVIEW

File: .github/workflows/static-test.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  static-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+      with:
+        # Use the PR branch as-is, not the PR branch merged in master
+        # (default behavior in GH actions)
+        ref: ${{ github.head_ref }}
```
Comment (aabadie) [inline .github/workflows/static-test.yml:21]: So apparently this doesn't work from a fork...

**Comment by aabadie** [comment]
Ok this is time it is good, yeah! Ready for a review :)

**Comment by fjmolinas** [comment]
Can you squash @aabadie?

**Comment by aabadie** [comment]
squashed, static tests should be green now.

**Review by fjmolinas — APPROVED** [review]
+1 for migrating to github actions, I would rather use a single tool. Looked at the test runs it seems to be working fine. It also seems to be a bit faster ~4m vs ~6m. ACK.

**Pull request merged**